framework/src/binding/store/primitive.rs

```diff
@@ -77,132 +77,138 @@ impl<S: ServiceState> DefaultStoreUint64<S> {
 
     // Self minus val
     // And set the result back to self
-    fn sub_(&mut self, val: u64) -> ProtocolResult<()> {
-        let sv = self.get_()?;
+    fn inner_sub(&mut self, val: u64) -> ProtocolResult<()> {
+        let sv = self.inner_get()?;
 
         if sv >= val {
-            self.set_(sv - val)
+            self.inner_set(sv - val)
         } else {
             Err(StoreError::Overflow.into())
         }
     }
 
     // Multiply val with self
     // And set the result back to self
-    fn mul_(&mut self, val: u64) -> ProtocolResult<()> {
-        let sv = self.get_()?;
+    fn inner_mul(&mut self, val: u64) -> ProtocolResult<()> {
+        let sv = self.inner_get()?;
 
         match val.overflowing_mul(sv) {
-            (mul, false) => self.set_(mul),
+            (mul, false) => self.inner_set(mul),
             _ => Err(StoreError::Overflow.into()),
         }
     }
 
     // Power of self
     // And set the result back to self
-    fn pow_(&mut self, val: u32) -> ProtocolResult<()> {
-        let sv = self.get_()?;
+    fn inner_pow(&mut self, val: u32) -> ProtocolResult<()> {
+        let sv = self.inner_get()?;
 
         match sv.overflowing_pow(val) {
-            (pow, false) => self.set_(pow),
+            (pow, false) => self.inner_set(pow),
             _ => Err(StoreError::Overflow.into()),
         }
     }
 
     // Self divided by val
     // And set the result back to self
-    fn div_(&mut self, val: u64) -> ProtocolResult<()> {
-        let sv = self.get_()?;
+    fn inner_div(&mut self, val: u64) -> ProtocolResult<()> {
+        let sv = self.inner_get()?;
 
         if let 0 = val {
             Err(StoreError::Overflow.into())
         } else {
-            self.set_(sv / val)
+            self.inner_set(sv / val)
         }
     }
 
     // Remainder of self
     // And set the result back to self
-    fn rem_(&mut self, val: u64) -> ProtocolResult<()> {
-        let sv = self.get_()?;
+    fn inner_rem(&mut self, val: u64) -> ProtocolResult<()> {
+        let sv = self.inner_get()?;
 
         if let 0 = val {
             Err(StoreError::Overflow.into())
         } else {
-            self.set_(sv % val)
+            self.inner_set(sv % val)
         }
     }
 }
 
 impl<S: ServiceState> StoreUint64 for DefaultStoreUint64<S> {
     fn get(&self) -> u64 {
-        self.get_()
+        self.inner_get()
             .unwrap_or_else(|e| panic!("StoreUint64 get failed: {}", e))
     }
 
     fn set(&mut self, val: u64) {
-        self.set_(val)
+        self.inner_set(val)
             .unwrap_or_else(|e| panic!("StoreUint64 set failed: {}", e));
     }
 
     // Add val with self
     // And set the result back to self
-    fn add(&mut self, val: u64) {
-        self.add_(val)
-            .unwrap_or_else(|e| panic!("StoreUint64 add value failed: {}", e));
+    fn add(&mut self, val: u64) -> bool {
+        let mut overflow = false;
+        self.inner_add(val).unwrap_or_else(|_| overflow = true);
```

Comment:
Consider the case where an error occurred other than an overflow, such as IO.

Reply:
fixed